content/SDKs/javascript/api-reference.mdx

```diff
@@ -260,6 +263,129 @@ const deletedCount = await context.vector.delete('product-descriptions', 'id1');
 console.log(`Deleted ${deletedCount} vector(s)`);
 ```
 
+### Object Storage
+
+The Object Storage API provides a way to store and retrieve objects (files, documents, media) with support for public URL generation. It is accessed through the `context.objectstore` object.
+
+#### `get(bucket: string, key: string): Promise<DataResult>`
+
+Retrieves an object from the object storage.
+
+##### Parameters
+
+- `bucket`: The bucket to get the object from
+- `key`: The key of the object to get
+
+##### Return Value
+
+Returns a Promise that resolves to a DataResult object. The DataResult has an `exists` boolean property and a `data` property containing the object data if found.
+
+##### Example
+
+```typescript
+// Retrieve an object from object storage
+const result = await context.objectstore.get('user-uploads', 'profile-123.jpg');
+if (result.exists) {
+  // Access data using the Data interface methods
+  const imageData = await result.data.binary();
+  console.log(`Image size: ${imageData.byteLength} bytes`);
+} else {
+  console.log('Image not found');
+}
+```
+
+#### `put(bucket: string, key: string, data: DataType, params?: ObjectStorePutParams): Promise<void>`
+
+Stores an object in the object storage.
+
+##### Parameters
+
+- `bucket`: The bucket to put the object into
+- `key`: The key of the object to put
+- `data`: The data to store (can be ArrayBuffer, string, or other DataType)
+- `params` (optional): Additional parameters including content type and encoding
+
+##### Return Value
+
+Returns a Promise that resolves when the object has been stored.
+
+##### Example
+
+```typescript
+// Store a text file
+await context.objectstore.put('documents', 'readme.txt', 'Hello, world!', {
+  contentType: 'text/plain'
+});
+
+// Store a binary file
+const imageData = new Uint8Array([/* image bytes */]).buffer;
+await context.objectstore.put('images', 'photo.jpg', imageData, {
+  contentType: 'image/jpeg'
+});
+
+// Store with custom encoding
+await context.objectstore.put('compressed', 'data.gz', compressedData, {
+  contentType: 'application/octet-stream',
+  contentEncoding: 'gzip'
+});
+```
+
+#### `delete(bucket: string, key: string): Promise<boolean>`
+
+Deletes an object from the object storage.
+
+##### Parameters
+
+- `bucket`: The bucket to delete the object from
+- `key`: The key of the object to delete
+
+##### Return Value
+
+Returns a Promise that resolves to a boolean indicating whether the object was deleted (true) or didn't exist (false).
+
+##### Example
+
+```typescript
+// Delete an object
+const wasDeleted = await context.objectstore.delete('user-uploads', 'old-file.pdf');
+if (wasDeleted) {
+  console.log('File deleted successfully');
+} else {
+  console.log('File did not exist');
+}
+```
+
+#### `createPublicURL(bucket: string, key: string, expiresDuration?: number): Promise<string>`
+
+Creates a time-limited public URL for accessing an object.
```

Comment:
we should discuss the default expiration if not provided which is 1 hour

if less than 1 minute, the expiration will be set to 1 minute.